app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/JsonCellPlotGenericController.java

```diff
@@ -0,0 +1,33 @@
+package uk.ac.ebi.atlas.experimentpage.cellplot;
+
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import static uk.ac.ebi.atlas.utils.GsonProvider.GSON;
+
+@RestController
+@RequestMapping(value="/json/cell-plots/generic/{experimentAccession}", method= RequestMethod.GET)
+public class JsonCellPlotGenericController {
+    private final CellPlotGenericDao cellPlotGenericDao;
+
+    public JsonCellPlotGenericController(CellPlotGenericDao cellPlotGenericDao) {
+        this.cellPlotGenericDao = cellPlotGenericDao;
```

Comment:
Is there any reason to skip the service layer and inject DAO class straight into the controller?

of course, I would like to know more about this concept of Generic implementation as I'm on holiday :).